### PR TITLE
moves a number of parameters to config file and fixes tests

### LINKED ISSUE #99: YouTube API key as config option
I had some issues getting the search feature to work, and finally solved it by changing the youtube API key directly in the backend.py script, but i think it would be easier and also would help others facing the same problem if you coul change the API key from the config files .
Something like this:

`[youtube]`
`api_key = '<key_goes_here>'`

### PR DESCRIPTION
**Pull request by natumbri**
makes each of the following  configuration options:
search_results = 15
playlist_max_videos = 20
youtube_api_key =
threads_max = 2

also records fresh cassettes for testing and fixes up other minor issues with tests

closes #99

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ before_install:
 
 install:
   - "pip install tox"
+  - "sudo -H pip install youtube-dl"
```
Comment (brammittendorff-dd) [inline .travis.yml]: Why would you execute this as sudo?

Comment (natumbri) [reply .travis.yml]: Perhaps it is because the script only needs youtube-dl as a dependency of pafy, and doesn't import youtube-dl directly?

Comment (brammittendorff-dd) [reply .travis.yml]: @natumbri hmm could me let met check this :smile:

Comment (natumbri) [reply .travis.yml]: Thanks! Sorry for all the commits that keep coming through - I haven't really gotten the hang of git yet, but I'm working it out, hopefully...

**Comment by natumbri** [comment]
Is there any way I can move these last 4 to a different place? They don't relate to the config options change...

**Comment by kingosticks** [comment]
You can, yes. Assuming you have all your changes in a branch called `feature/api-key-setting`:
```
git checkout -b feature/unrelated-stuff
git checkout feature/api-key-setting
git reset --hard HEAD^^^^
git push --force
```

The very last thing I want to do here is dampen your enthusiasm but can I just reiterate the importance of keeping PRs tightly focused. Adding just the API key setting could have been a few lines of changes which could be reviewed easily and merged quickly. We could have done the others separately. And the general test fixing separately too (assuming that's not directly relating to those new settings). Doing each of these tasks in a separate git branch means you can still work on them all at the same time so it won't block your progress. That type of work flow is very common and it's what we recommend in our development docs. I know all too well how things can snowball into larger changes and sometimes it's unavoidable but it's really helpful if you can resist it. 

As it is, for example, I can't review this on my phone so it'll have to wait.

**Comment by natumbri** [comment]
I totally see where you are coming from, and agree!  The main problem I'm having is learning how to use git.  I really meant to keep these things separate, but because of the way I used Git, I managed to mash them all together, and didn't really know how to unmash them.  I will try to pull it back to just the settings and the tests.  And will try hard to use branches to keep the tasks separate.  

One of the difficult things for me (apart from not knowing about how to do this stuff, generally) is that I am now trying to reverse engineer incremental changes from code that has move significantly away from where it started.

Anyway, thanks for your patience and assistance - hopefully we can get to a point where I can introduce the non-API based code, so that this can then move from a module that doesn't work, to one that does!

File: .travis.yml (hunk 1 above)
Comment (natumbri) [reply .travis.yml]: I think "install:" was the wrong place for youtube-dl, which is why teh sudo was necessary. 
 This works: https://github.com/mopidy/mopidy-youtube/commit/d25e57529eb56bb7f58ca59e87b9807e48fb353b

**Pull request closed without merging**